Assets/Scripts/Utils.cs

```diff
@@ -1,38 +1,47 @@
 ﻿using UnityEngine;
-
-using System;
 using System.Collections.Generic;
+using System.Linq;
 using Random = UnityEngine.Random;
 
-class WeightedRandomBag<T>
+public class WeightedRandomBag<T>
 {
     private struct Entry
     {
-        public float accumulatedWeight;
-        public T item;
+        public float AccumulatedWeight;
+        public T Item;
     }
 
-    private List<Entry> entries = new List<Entry>();
+    private readonly List<Entry> entries = new List<Entry>();
     private float accumulatedWeight;
 
     public void AddEntry(T item, float weight)
     {
         accumulatedWeight += weight;
-        entries.Add(new Entry {item = item, accumulatedWeight = accumulatedWeight});
+        entries.Add(new Entry {Item = item, AccumulatedWeight = accumulatedWeight});
     }
 
     public T GetRandom()
     {
         var r = Random.value * accumulatedWeight;
 
-        foreach (var entry in entries)
+        foreach (var entry in entries.Where(entry => entry.AccumulatedWeight >= r))
```

Comment:
You could go full LINQ and do a Select at the end that's like `.Where(...).Select(e => e.Item)`

Reply:
I one-upped you